DigitalLearningSolutions.Web/Controllers/Support/ResourcesController.cs

```diff
@@ -0,0 +1,66 @@
+﻿namespace DigitalLearningSolutions.Web.Controllers.Support
+{
+    using System.Threading.Tasks;
+    using DigitalLearningSolutions.Data.Enums;
+    using DigitalLearningSolutions.Data.Helpers;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Attributes;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.Models.Enums;
+    using DigitalLearningSolutions.Web.ViewModels.Support.Resources;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.FeatureManagement;
+
+    [Route("/{dlsSubApplication}/Resources")]
+    [Authorize(Policy = CustomPolicies.UserCentreAdminOrFrameworksAdmin)]
+    [SetDlsSubApplication]
+    [SetSelectedTab(nameof(NavMenuTab.Support))]
+    public class ResourcesController : Controller
+    {
+        private readonly IConfiguration configuration;
+        private readonly IFeatureManager featureManager;
+        private readonly IResourcesService resourcesService;
+
+        public ResourcesController(
+            IFeatureManager featureManager,
+            IConfiguration configuration,
+            IResourcesService resourcesService
+        )
+        {
+            this.featureManager = featureManager;
+            this.configuration = configuration;
+            this.resourcesService = resourcesService;
+        }
+
+        public async Task<IActionResult> Index(DlsSubApplication dlsSubApplication)
+        {
+            if (!DlsSubApplication.TrackingSystem.Equals(dlsSubApplication) &&
+                !DlsSubApplication.Frameworks.Equals(dlsSubApplication))
+            {
+                return NotFound();
+            }
+
+            var trackingSystemSupportEnabled =
+                DlsSubApplication.TrackingSystem.Equals(dlsSubApplication) &&
+                User.HasCentreAdminPermissions() &&
+                await featureManager.IsEnabledAsync(FeatureFlags.RefactoredTrackingSystem);
+            var frameworksSupportEnabled = DlsSubApplication.Frameworks.Equals(dlsSubApplication) &&
+                                           User.HasFrameworksAdminPermissions();
+
+            if (trackingSystemSupportEnabled || frameworksSupportEnabled)
+            {
+                var model = new ResourcesViewModel(
+                    dlsSubApplication,
+                    SupportPage.Resources,
+                    configuration.GetCurrentSystemBaseUrl(),
+                    resourcesService.GetAllResources()
+                );
+                return View("Index", model);
+            }
+
+            return RedirectToAction("Index", "Home");

```

Comment:
Is it worth doing all this here? The access denied change will come automatically with the change to using the ValidateAllowed...Attribute.

The 404 is a bit more difficult as we have an extra condition that it can be enabled by each feature flag separately. 404 is the behaviour we get when we use the feature flag attribute normally, but I don't think we can this twice so we probably need something custom to deal with this. This has a few options depending on the behaviour we want:

- Have the same 404 check in the actions for ResourcesController and SupportController (or one of these if we don't want it in both)
- Have some new attribute that will check application + feature flag and return the 404
- Roll this check into the Validate...Attribute. This will potentially be useful for other pages that have the same pattern. E.g. MyAccount can probably currently be accessed at /TrackingSystem/MyAccount since I don't think we have a similar flag check there.

Reply:
Yeah, I think rolling it into the Validate...Attribute makes the most sense. Yeah, let's not implement this here, but leave todos here referencing 608 and comment on the tickets that we'll be testing these cases in 608. Would you mind doing that?

Reply:
I've added this to the ACs on 608.